libraries/botframework-streaming/src/namedPipe/namedPipeServer.ts

```diff
@@ -58,42 +58,42 @@ export class NamedPipeServer implements IStreamingTransportServer {
     /**
      * Used to establish the connection used by this server and begin listening for incoming messages.
      *
+     * @param callback Optional callback parameter that fires when server is listening on incoming/outgoing pipes
      * @returns A promised string that will not resolve as long as the server is running.
      */
-    public async start(): Promise<string> {
+    public async start(callback?: () => void): Promise<string> {
         const { PipePath, ServerIncomingPath, ServerOutgoingPath } = NamedPipeTransport;
 
         // The first promise resolves as soon as the server is listening. The second resolves when the server
-        // closes, or an error occurs. We want to ensure we are listening to the servers in series so that, if two
-        // processes start at the same time, only one is able to listen on both the incoming and outgoing sockets.
-        // We await the first promise to that we only proceed once we have the socket.
-        const incoming = await new Promise<Promise<void>>((resolveListening, rejectListening) => {
-            return new Promise<void>((resolveClosed, rejectClosed) => {
-                const server = createNodeServer((socket) => {
-                    if (this._receiver.isConnected) {
-                        return;
-                    }
-
-                    this._receiver.connect(new NamedPipeTransport(socket));
-                });
-
-                this._incomingServer = server;
+        // closes, or an error occurs. Wrapping with an array ensures the initial await only waits for the listening
+        // promise.
+        //
+        // We want to ensure we are listening to the servers in series so that, if two processes start at the same
+        // time, only one is able to listen on both the incoming and outgoing sockets.
+        const [incoming] = await new Promise<[Promise<void>]>((resolve, reject) => {
+            this._incomingServer = createNodeServer((socket) => {
+                if (this._receiver.isConnected) {
+                    return;
+                }
 
-                server.once('closed', resolveClosed);
+                this._receiver.connect(new NamedPipeTransport(socket));
+            }).once('error', reject);
 
-                server.once('error', (err) => {
-                    rejectListening(err);
-                    rejectClosed(err);
-                });
+            const listening = new Promise<void>((resolve, reject) => {
+                this._incomingServer.once('closed', resolve);
+                this._incomingServer.once('error', reject);

```

Comment:
tiny nit: it would be easier on the eyes if the `resolve` and `reject` of the inner promise had different names than the outer promise.

(not blocking)